### ISSUE TITLE
POPaddress becomes "undefined"

### ISSUE THREAD

**Issue by despotak** [TrainerRoshi/Roshi-HHeroes-Trainer#53]
Starting the "Places of Power" on HentaiHeroes.com (not in ComixHarem.com neither PornstarHarem.com to my knowledge) from a clean slate (all PoP rewards collected) goes though the list adding girls to the quests and when all PoPs are populated the `POPaddress` becomes `undefined` and the script falls into a 10 second loop.

`POPaddress`  takes a valid value again only after I claim at least one PoP reward, but falls into the same loop when all PoPs are populated again.

Tried to delete the local storage but it did not resolve the problem.

**System**
**OS:** Windows 11, build 22623.875
**Browser:** Microsoft Edge 108.0.1462.2
**Userscript Manager:** Tampermonkey 4.16.1 (latest for edge. `4.18.0` is not available yet)
**Active Userscripts:** 
- Master Roshi's HHeroes Trainer v. 1.60
- Hentai Heroes++ BDSM version v. 1.28.0 (do not condemn me master)

[**Console Log**](https://gist.github.com/despotak/7ea91439cc0f2404ac77b751d79dfef3) (I can provide a verbose log if needed, after my current PoP run is finished)

**Comment by TrainerRoshi** [comment]
Try v1.61 and see if the problem persists. In the log you included (♥!!!) I can see that Auto Activities interrupted Auto PoP and gave it a nonsense value. If you delete the cookie `nextPOPTime` and the local storage `POPaddress` it should work. 

My script has gotten messy. I need to clean up the console messages and generally make every function fool-proof against errors. I catch most errors in try statements but not all.

**Comment by despotak** [comment]
The `undefined ` problem is solved, but the infinite loop is still there.

I suspect that you have something `while (Number_of_available_girls != 0) do setCookie(nextPOPTime, 7hrs) `. 

Mine was apparently an edge case, were the POP levels were low enough for me to have enough girls to populate all the spots and still have girls remaining, so that the loop never ends. Now that those POPs leveled up a bit and I need to use all my girls I have no problems.

### Other notes

v1.61 broke the "Salary collect-all button" functionality. v1.60 was collecting immediately when any amount was available (not as intended according to the changelog), but now it does not collect at all, even after the sum exceeds 100.000.

The icon and menu for the monthly pass card gets hidden. Is this intentional?

New world (and troll?) for HH added.

Now that ★MAX★ and ★SUPERMAX★ are not a thing anymore, does the secret do anything? (except disabling the very useful pause functionality)

**Comment by TrainerRoshi** [comment]
Ok I fixed the collect all button. Thanks for letting me know. I never see that button so I rely on you guys to let me know if it breaks.

A long time ago I added some kind of check for players without enough girls to do Places of Power. I'll check it and see how it broke. My ten second delay in PoP is just so the game finishes refreshing it's data variables.

I hid the useless treasure chests and credit card icons at the bottom of the home page. I didn't realize that also hid the menu in the calendar area. I'll fix that.

New troll and world!? Yay!

New secret soon in a different place; stay tuned on youtube.
